astropy/visualization/wcsaxes/tests/test_coordinate_helpers.py

```diff
@@ -205,3 +212,53 @@ def test_set_major_formatter():
 
     ax.coords[1].set_major_formatter("dd:mm:ss.s", show_decimal_unit=False)
     assert ax.coords[1].format_coord(4) == "4\xb000'00.0\""
+
+
+def test_set_position_invalid():
+    fig = Figure()
+    _canvas = FigureCanvasAgg(fig)
+    ax = WCSAxes(fig, [0.1, 0.1, 0.8, 0.8], aspect="equal")
+    fig.add_axes(ax)
+
+    with pytest.warns(
+        AstropyDeprecationWarning,
+        match=r"Ignoring unrecognized position\(s\): \['x'\], should be one of b/r/t/l",
+    ):
+        ax.coords[0].set_ticks_position("xl")
+    with pytest.warns(
+        AstropyDeprecationWarning,
+        match=r"Ignoring unrecognized position\(s\): \['o'\], should be one of b/r/t/l",
+    ):
+        ax.coords[1].set_ticklabel_position("to")
+    with pytest.warns(
+        AstropyDeprecationWarning,
+        match=r"Ignoring unrecognized position\(s\): \['q', 'p'\], should be one of b/r/t/l",
+    ):
+        ax.coords[1].set_axislabel_position("qbp")
+
+    assert ax.coords[0].get_ticks_position() == ["l"]
+    assert ax.coords[1].get_ticklabel_position() == ["t"]
+    assert ax.coords[1].get_axislabel_position() == ["b"]
+
+
+def test_set_position_invalid_gridline():
+    fig = Figure()
+    _canvas = FigureCanvasAgg(fig)
+    ax = WCSAxes(fig, [0.1, 0.1, 0.8, 0.8], aspect="equal")
+    fig.add_axes(ax)
+
+    ax.coords[0].add_tickable_gridline("my-grid-line", -30 * u.one)
+
+    with pytest.warns(
+        AstropyDeprecationWarning,
+        match=r"Ignoring unrecognized position\(s\): \['my-parrot-line'\]",
+    ):
+        ax.coords[1].set_ticks_position(["my-grid-line", "my-parrot-line"])
+
+    with pytest.warns(
+        AstropyDeprecationWarning,
+        match=r"It looks like 'my-grid-line' matches the name of a single axis. If "
+        r"you are trying to specify a multi-character axis name, use a list "
+        r"or a tuple, e.g. \('my-grid-line',\).",
```

Comment:
Is this a change in behaviour? Or was this always the preferred way?

I am never a massive fan of length-1 tuples.

Reply:
I'm pretty sure it's not a change in behavior - before we were silently interpreting strings one character at a time so 'my-grid-line' would have matched 'l' (left)  and 'r' (right) in a rectangular frame, and the other letters would have been ignored.

We can't easily support multi-character strings here because of the ambiguity with strings like 'lbrt' and so on. If we did want to change the behavior somehow we should do this in another PR.